### PR TITLE
#13 header component

### LINKED ISSUE #13: Create Header component
**User Story**
As a user, I want to see a header component that contains important components: date time, search bar, button to add widgets.

**Acceptance Criteria**
* A header component that sits at the top of the page
* Component contains Focus logo, date and time, search bar, button to add widgets, and profile settings (see Figma design linked below)
A header component that will contain the Focus logo, search bar, date and time, profile button

**Additional Context**
Figma Link: https://www.figma.com/file/J2DDVa9redQigCzTee56xw/SE701-Daily-Focus?node-id=0%3A1

**Dependencies**
Search bar component: #7 
Date time component: #4

### PR DESCRIPTION
**Pull request by sannan06**
**Issue:**
Resolves #13 

**Solution:**
Created Header component to specifications in Figma prototype. Included SearchBar component from #7 

**Test Method:** 
Manual test on firefox & safari

**Risk:**
Changes content of App.js & shows Header + empty body as new home page

**Reviewed By:**
[Edit the message as to who reviewed it here]

### CODE REVIEW

File: client/src/components/SearchBar/index.js [hunk 1]
```diff
@@ -1,32 +1,57 @@
...
 import Paper from "@material-ui/core/Paper";
 import InputBase from "@material-ui/core/InputBase";
-import IconButton from "@material-ui/core/IconButton";
+import { IconButton, makeStyles } from "@material-ui/core";
 import SearchIcon from "@material-ui/icons/Search";
 
+import styles from "./style.module.scss";
+import GoogleLogo from "../../images/google-icon.svg";
+
+const useStyles = makeStyles({
```
[the remark this answers is not in the document]
Comment (sannan06) [reply client/src/components/SearchBar/index.js:10]: Material UI components have bespoke APIs for styling them, especially given they can contain nested components within them. If I need to e.g. access the InputLabel within an InputBase, I would have to target the `input` class on the InputLabel component. If you set the `className` and try to style it from the `SCSS` file, you will be fighting against CSS specificity where the default material-ui element styling will take precedence over your styling & force you to litter your stylesheet with `!important` tags. 

More on the subject: https://material-ui.com/customization/components/

[the remark this answers is not in the document]
Comment (Denijar) [reply client/src/components/SearchBar/index.js:10]: Awesome okay thanks for the explanation

**Comment by Denijar** [comment]
So ideally we have unit tests accompanying PRs (I'll need to retroactively add tests for the searchbar because I missed that requirement - so let's try not have any more PRs go through without tests). Once #46 is resolved, I think a snapshot test would be sufficient for this header component.

File: client/src/components/Header/index.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
+import React from "react";
+import Button from "@material-ui/core/Button";
+import { IconButton, makeStyles } from "@material-ui/core";
+import PersonIcon from "@material-ui/icons/Person";
+import styles from "./style.module.scss";
+import FocusLogo from "../../images/focus-logo.svg";
+import SearchBar from "../SearchBar";
```
Comment (Denijar) [inline client/src/components/Header/index.js:7]: Would it be cleaner to nest the SearchBar component folder inside the Header folder or do you think that is unnecessary?

**Review by Chuppay — APPROVED** [review]
LGTM

File: client/src/components/Header/index.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+const useStyles = makeStyles({
```
[the remark this answers is not in the document]
Comment (sannan06) [reply client/src/components/Header/index.js:9]: These styles ares styling the material-ui components, whereas the styling in the SCSS file is styling the rest of the elements & handling their positioning. I don't think the `makeStyles` API can be used for styling non material-ui elements?

**Review by Denijar — APPROVED** [review]
#46 was a blocker to merging this PR, but #46 is blocked due to package incompatibility errors. I think the best option is to merge this PR. This will allow me to make a follow up PR to add test cases for the header and the search bar (which I can't do separately because this PR makes changes to SearchBar). I will also create some simple documentation about testing.

**Pull request merged**